integration/test/Test/AssetDownload.hs

```diff
@@ -0,0 +1,70 @@
+module Test.AssetDownload where
+
+import API.Cargohold
+import qualified Data.ByteString.Char8 as C
+import GHC.Stack
+import Network.HTTP.Client (Request (redirectCount))
+import qualified Network.HTTP.Client as HTTP
+import Network.HTTP.Types
+import Network.URI
+import SetupHelpers
+import Testlib.Prelude
+
+testDownloadAsset :: HasCallStack => App ()
+testDownloadAsset = do
+  user <- randomUser OwnDomain def
+
+  key <- bindResponse (uploadAsset user) $ \resp -> do
+    resp.status `shouldMatchInt` 201
+    resp.json %. "key"
+
+  void . bindResponse (downloadAsset user key id) $ \resp -> do
+    resp.status `shouldMatchInt` 200
+    assertBool
+      ("Expect 'Hello World!' as text asset content. Got: " ++ show resp.body)
+      (resp.body == fromString "Hello World!")
+
+testDownloadAssetMultiIngressS3DownloadUrl :: HasCallStack => App ()
+testDownloadAssetMultiIngressS3DownloadUrl = do
+  user <- randomUser OwnDomain def
+
+  key <- bindResponse (uploadAsset user) $ \resp -> do
+    resp.status `shouldMatchInt` 201
+    resp.json %. "key"
+
+  withModifiedService
+    Cargohold
+    modifyConfig
+    $ do
+      bindResponse (downloadAsset user key noRedirects) $ \resp -> do
+        resp.status `shouldMatchInt` 404
+      bindResponse (downloadAsset' user key "red.example.com" noRedirects) $ \resp -> do
+        resp.status `shouldMatchInt` 302
+        locationHeaderHost resp `shouldMatch` "s3-download.red.example.com"
+      bindResponse (downloadAsset' user key "green.example.com" noRedirects) $ \resp -> do
+        resp.status `shouldMatchInt` 302
+        locationHeaderHost resp `shouldMatch` "s3-download.green.example.com"
+      bindResponse (downloadAsset' user key "unknown.example.com" noRedirects) $ \resp -> do
+        resp.status `shouldMatchInt` 404
+        resp.json %. "label" `shouldMatch` "no-asset-endpoint"
```

Comment:
```suggestion
        resp.json %. "label" `shouldMatch` "not-found"
```

also test label in the block in line 40 above.  this way you don't leak the fact that the server has a multi-ingress setup.

Reply:
Yep, that's a good idea :+1: